proto/regen/ecocredit/v1/tx.proto

```diff
@@ -165,6 +179,38 @@ message MsgCreateBatchResponse {
   string batch_denom = 1;
 }
 
+// MsgMintBatch is a request type for MintBatch Msg RPC.
+message MsgMintBatch {
+  // Issuer must equal to the batch.issuer address.
+  // Signer of the msg.
+  string issuer = 1;
+
+  // issuance are the credits issued in the batch.
+  repeated BatchIssuance issuance = 2;
```

Comment:
If there are multiple `BatchIssuance`s for a given `MsgMintBatch` is it expected that they all have the same `origin_tx`?

We could alternatively evaluate nesting `origin_tx` inside BatchIssuance message, so that an on-chain minting event can batch together multiple issuances (coming from multiple different bridge transactions). This may be suboptimal though, as leveraging events becomes quite cumbersome and annoying if in the current event system if we have multiple event attributes corresponding to different origin bridge transactions for one SDK tx.

Also- do we need to support `origin_tx` information in `MsgCreateBatch()` in cases where a bridge action triggers a new batch creation? Or do we expect that the bridge creator does not mint any credits when they create a new batch for a bridge (so all minting of credits via a bridge always happen through a `MsgMintBatch()` call).

I see a few options:

### All issuances from bridges must happen through `MsgMintBatch()` 

In this case, we don't need `origin_tx` information in `MsgCreateBatch()`, and we should make sure we have some on-chain validation that `MsgCreateBatch()` fails if `sealed == false` and `issuance != nil`.

### Allow issuances from bridges in `MsgCreateBatch()`

In this case, we should add `origin_tx` into `MsgCreateBatch()`

Reply:
> Also- do we need to support origin_tx information in MsgCreateBatch() in cases where a bridge action triggers a new batch creation?

Good point. yes!

Reply:
I've updated the `MsgCreatBatch` - so bridge will create a batch and mint at the same time. It will be consistent with the minting we have now. Subsequent mints will use MsgMintBatch